lib/netsnmp/message.rb

```diff
@@ -35,7 +35,8 @@ def decode(stream, security_parameters:)
       _, _, message_flags, = headers.value
 
       # get last byte
-      security_level = message_flags.with_label(:message_flags).value.unpack("C*").last
+      # Cisco ASA send an extra byte on the security_level
+      security_level = message_flags.with_label(:message_flags).value.unpack("C*").last % 8
```

Comment:
I'm afraid this is not correct. The [msg flags from a v3 message is defined as the last byte](http://www.tcpipguide.com/free/t_SNMPVersion3SNMPv3MessageFormat.htm), for which one has to get the last 2 bits. The error is that I didn't take into account that (for some reason) agents might use the remaining.

The real fix here is to discard the left-outermost bits and keep the remaining two, i.e. do the following:

```ruby
security_level = message_flags.with_label(:message_flags).value.unpack("C*").last % 3
```

Let me know if you have further questions.

Reply:
Hello, thank you for the explanation.

I'm not familiar with the snmp protocol, but I updated the PR with your fix and it's working well :)